### ISSUE TITLE
Java 7 build

### ISSUE THREAD

**Issue by headwinds** [pugnascotia/spring-cra-demo#1]
You list Java 8 as being required... just wondering if it's a hard requirement... could this work with Java 7?

I followed your nice tutorial and arrived this screen when when I hit the localhost:8080

```
Whitelabel Error Page

This application has no explicit mapping for /error, so you are seeing this as a fallback.

Wed May 10 09:11:58 EDT 2017
There was an unexpected error (type=Not Found, status=404).
No message available
```

Also had to add this to my pom.xml
```
<plugin>
     <groupId>org.apache.maven.plugins</groupId>
     <artifactId>maven-compiler-plugin</artifactId>
     <version>3.0</version>
     <configuration>
          <source>1.6</source>
              <target>1.6</target>
     </configuration>
</plugin>
```
then built it by 
```
mvn clean install -X
java -jar target/cra-demo-0.0.1-SNAPSHOT.jar
```
I'm coming at this project from primarily UI background - I'm working with a team of Java developers and want to introduce it to the team - problem being were stuck on Java 7 for the time being...

**Comment by brandonflowers** [comment]
I've learned a bit more java now including how to [switch](https://wimdeblauwe.wordpress.com/2014/03/20/switching-easily-between-java-jdks-on-mac-os-x/) between different installed versions. 

I tried this tutorial again and I still get the whitelabel error page.  Now that I'm sure I have java 8 (jdk 1.8) installed and have reverted my change to the pom with a fresh directory, it should work eh?!

**Comment by headwinds** [comment]
I've learned a bit more java now including how to [switch](https://wimdeblauwe.wordpress.com/2014/03/20/switching-easily-between-java-jdks-on-mac-os-x/) between different installed versions. 

I tried this tutorial again and I still get the whitelabel error page.  Now that I'm sure I have java 8 (jdk 1.8) installed and have reverted my change to the pom pulling it into a fresh directory, it should work eh?!

**Comment by leonroy** [comment]
Same issue here, don't think it's a Java problem.

A quick fix is to edit `pom.xml` and change `							<outputDirectory>${basedir}/target/classes</outputDirectory>` to `                         <outputDirectory>${basedir}/target/classes/static</outputDirectory>`.

**Comment by brandonflowers** [comment]
@leonroy so you can build both java and create react app with maven using this fix to the pom? I ended up abandoning maven and came up with [this proxy approach](https://medium.com/@headwinds/opposing-corners-how-create-react-app-and-java-spring-security-can-talk-through-a-proxy-dda3c2e08a04) instead

**Comment by headwinds** [comment]
@leonroy so you can build both java and create react app with maven using this fix to the pom? I ended up abandoning maven and came up with [this proxy approach](https://medium.com/@headwinds/opposing-corners-how-create-react-app-and-java-spring-security-can-talk-through-a-proxy-dda3c2e08a04) instead

**Comment by leonroy** [comment]
@headwinds yep, builds fine. I'm coming to the same conclusion - feeling a single page app wrapped in a Spring Boot/Maven container is pretty pointless unless you plan on using the other features of Spring Boot.

It's nice for build purposes but yeah, think a proxy approach makes more sense.

**Comment by pugnascotia** [comment]
Closed by #2
